src/Appwrite/Platform/Tasks/Restore.php

```diff
@@ -0,0 +1,291 @@
+                    Console::error($stderr);
+                    Console::exit();
+                }
+            }
+
+            $files = $local->getRoot() . '/files';
+
+            Backup::log('Creating Directory: ' . $files);
+            if (!mkdir($files, 0755, true)) {
+                Console::error('Error creating directory: ' . $files);
+                Console::exit();
+            }
+
+            $this->download($filename, $local);
+            $this->untar($local->getPath($filename), $files);
+        } else {
+            $local = new Local(Backup::BACKUPS . '/' . $database . '/full/' . $id);
+            $files = $local->getRoot() . '/files';
+        }
+
+        if (!file_exists($files)) {
+            Console::error('Directory not found: ' . $files);
+            Console::exit();
+        }
+
+        $this->decompress($files);
+        $this->prepare($files);
+        $this->restore($files, $cloud, $datadir);
+
+        Backup::log("Restore Finish in " . (microtime(true) - $start) . " seconds");
+    }
+
+    public function download(string $filename, Device $local)
+    {
+        $s3 = new DOSpaces($this->database . '/full', App::getEnv('_DO_SPACES_ACCESS_KEY'), App::getEnv('_DO_SPACES_SECRET_KEY'), App::getEnv('_DO_SPACES_BUCKET_NAME'), App::getEnv('_DO_SPACES_REGION'));
+
+        try {
+            $path = $s3->getPath($filename);
+
+            if (!$s3->exists($path)) {
+                Console::error('File: ' . $path . ' does not exist on cloud');
+                Console::exit();
+            }
+
+            $file = $local->getPath($filename);
+            Backup::log('Downloading: ' . $file);
+
+            if (!$s3->transfer($path, $file, $local)) {
+                Console::error('Error Downloading ' . $file);
+                Console::exit();
+            }
+        } catch (Exception $e) {
+            Console::error($e->getMessage());
+            Console::exit();
+        }
+    }
+
+    public function untar(string $file, string $directory)
+    {
+        $stdout = '';
+        $stderr = '';
+        $cmd = 'tar -xzf ' . $file . ' -C ' . $directory;
+        Backup::log($cmd);
+        Console::execute($cmd, '', $stdout, $stderr);
+        if (!empty($stderr)) {
+            Console::error($stderr);
+            Console::exit();
+        }
+
+        if (!file_exists($file)) {
+            Console::error('Restore file not found: ' . $file);
+            Console::exit();
+        }
+    }
+
+    public function decompress(string $target)
+    {
+        $logfile = $target . '/../decompress.log';
+
+        $args = [
+            '--user=' . $this->dsn->getUser(),
+            '--password=' . $this->dsn->getPassword(),
+            '--host=' . $this->dsn->getHost(),
+            '--decompress',
+            '--strict',
+            '--remove-original', // Removes *.lz4 compressed files
+            '--parallel=' . self::PROCESSORS,
+            '--compress-threads=' . self::PROCESSORS,
```

Comment:
Is this option relevant for decompression?

Reply:
As I understand this can make the decompression faster, I think later on will need to adjust the values for 
parallel & compress-threads PROCESSORS

Reply:
Maybe we could use the `nproc` command to set the value of PROCESSORS? That will output the number of processors available on the machine it was run on:

Reply:
Nice.. Updated..
@shimonewman Can you check this out on the cloud, please?